org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/snippets/CommentSnippetContext.java

```diff
@@ -36,6 +37,13 @@ public boolean isMatch(ICompletionRequest request, Map<String, String> model) {
 			// completion was triggered inside doctype declaration, ignore the snippets
 			return false;
 		}
+		
+		DOMDocument document = request.getXMLDocument();
+		int offset = request.getOffset();
+		if (document.hasProlog() && offset == 0){

```

Comment:
Not tested, but is it working if you have some spaces before the xml(declaration and offset is not equals to 0? I mean :

`       |       <?xml ?>`

Reply:
I think that would trigger, but I haven't given that much thought because it's marked as invalid anyways.  
Should I address it nonetheless?

Reply:
Yes we must manage this case even if validation report an error. For instance if user is typing quickly, he cannot see the error validation. Imagine that validation could be disable (it's not the case), snippet must be shown only if it's valid.

Reply:
done.  
But in these cases the `#region` and `#endregion` snippets trigger anyways. This is clearly wrong, but I didn't want to touch them because their implementation differs from all other snippets...

Reply:
We should switch #region and #endregion in snippets. Please create an issue for that and if you wish yo do that and assign you to the issue.

Reply:
ok, I open an issue for that.

Reply:
but for this PR, should I leave them in the unit test, so the test passes, even tough it is wrong? Or should I keep the test correct, which will than inevitably fail?

```Java
xml = "| <?xml version=\"1.0\" encoding=\"UTF-8\" ?>";
testCompletionFor(xml, REGION_SNIPPETS); // test passes, but it's wrong
```
or?

```Java
xml = "| <?xml version=\"1.0\" encoding=\"UTF-8\" ?>";
testCompletionFor(xml, 0); // test fails
```

Reply:
CI build must not fails because it build LemMinx JAR on each commit (and vscode-xml as well). The test will be fixed when region will be fixed too.